### PR TITLE
Improve the docs of no_input when using defaults

### PR DESCRIPTION
**Pull request by pokoli**
- Remove the exsitings TODOs
- Add an example.

### CODE REVIEW

File: docs/advanced_usage.rst [hunk 1]
```diff
@@ -170,12 +170,15 @@ To suppress the prompts asking for input, use `no_input`.
 Basic Example: Using the Defaults
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-TODO: document `no_input`:
+Cookiecutter will pick its defaults value if ussing with `no_input`::
```
Comment (michaeljoseph) [inline docs/advanced_usage.rst]: > Cookiecutter will pick **a** **default** value if **used** with `no_input

File: docs/advanced_usage.rst [hunk 2]
```diff
@@ -170,12 +170,15 @@ To suppress the prompts asking for input, use `no_input`.
...
-* As command-line argument
-* As parameter of `cookiecutter()`
+    from cookiecutter.main import cookiecutter
+    cookiecutter(
+        'cookiecutter-django',
+        no_input=True,
+    )
 
-TODO: document where context values come from in this example (`cookiecutter.json` and `.cookiecutterrc`)
+In this case it will be using the default defined in `cookiecutter.json` and `.cookiecutterrc`.
```
Comment (michaeljoseph) [inline docs/advanced_usage.rst]: I think we may need some concrete values here and to note that values from `.cookiecutterrc` will override values in `cookiecutter.json`.

**Comment by hackebrot** [comment]
🔔

**Comment by pokoli** [comment]
Let me know if you expect my to rebase and squash both commits in one. So when the patch is ready for mergin I will squash them in a single commit.

File: docs/advanced_usage.rst [hunk 3]
```diff
@@ -170,12 +170,18 @@ To suppress the prompts asking for input, use `no_input`.
...
 
-* As command-line argument
-* As parameter of `cookiecutter()`
+    from cookiecutter.main import cookiecutter
+    cookiecutter(
+        'cookiecutter-django',
+        no_input=True,
+    )
+
+In this case it will be using the default defined in `cookiecutter.json` and `.cookiecutterrc`.
```
Comment (michaeljoseph) [inline docs/advanced_usage.rst]: > defined in `cookiecutter.json` _or_ `.cookiecutterrc`

File: docs/advanced_usage.rst [hunk 4]
```diff
@@ -170,12 +170,18 @@ To suppress the prompts asking for input, use `no_input`.
...
 
-TODO: document where context values come from in this example (`cookiecutter.json` and `.cookiecutterrc`)
+.. note::
+   values from `.cookiecutterrc` will be overrided by values from `cookiecutter.json`
```
Comment (michaeljoseph) [inline docs/advanced_usage.rst]: It's the other way around 😜 
> `.cookiecutterrc` will be _overridden_ by ...

**Comment by michaeljoseph** [comment]
@pokoli Can you please rebase? #749 splits up the advanced usage docs.

**Comment by pokoli** [comment]
@michaeljoseph  Fixes comments and rebased!

**Comment by michaeljoseph** [comment]
Thanks @pokoli! 🍰

**Pull request merged**